tsdb/db.go

```diff
@@ -835,8 +835,12 @@ func (db *DB) compactHead(head *RangeHead) (err error) {
 	runtime.GC()
 
 	if err := db.reload(); err != nil {
-		if err := os.RemoveAll(filepath.Join(db.dir, uid.String())); err != nil {
-			return errors.Wrapf(err, "delete persisted head block after failed db reload:%s", uid)
+		var merr tsdb_errors.MultiError
+		merr.Add(err)
```

Comment:
So that we can distinguish the error (goes inside the next `if` as said in other comment)
```suggestion
		merr.Add(errors.Wrap(err, "reload blocks"))
```